### ISSUE TITLE
Example Code Not running

### ISSUE THREAD

**Issue by yves-weissenberger** [pavanramkumar/pyglmnet#127]
Hey,

I tried running your example code, but the outputs I get are very different to what is shown on the example page. I've tried on two separate machines.

Running python 2.7 with:

numpy (1.11.0)
scipy (0.17.0)
scikit-learn (0.17.1)

I've uploaded the run notebook [here](https://github.com/yves-weissenberger/misc/blob/master/plot_poisson_2.ipynb)

**Comment by jasmainak** [comment]
I can confirm the bug

**Comment by jasmainak** [comment]
I think it is related to this latest PR: https://github.com/pavanramkumar/pyglmnet/pull/123/. Also, we have an outstanding issue with random seeds which I just noticed has been closed. @pavanramkumar did we manage to fix that? It appears that we still have the seed problem.

**Comment by pavanramkumar** [comment]
Thanks @yves-weissenberger for reporting this issue! And thanks @jasmainak for first response.

To answer your question: yes, it is indeed related to PR #123 when modified the loss function to scale well with `n_samples`. As a results, the default learning_rate has increased. The API examples might not have been updated since then, but try using the default learning rates. 

In your example, you set `learning_rate=1e-5` which is very small for `max_iter=1000`. I tried using `2e-1`, `1e-1`, `1e-2` and get good results for all of them.

I know setting the learning rate correctly is kind of a pain and we're actively trying to avoid that in the future using Newton methods. 

Hope this helps! Please let us know if you're stuck anywhere.

**Comment by pavanramkumar** [comment]
@jasmainak I think the random seed issue has been fixed with an input parameter called `random_state`. 

At least, I tried re-running `fit()` and `predict()` for the same values of `Xr`, `yr`, `Xt`, `yt` and got the same results. 

You can also check that if you change `random_state` to a different value while instantiating the `GLM` object, you will get a different result, but that's not because the problem is non-convex, it's only because we get different values of `yr`, and `yt` when we call the `simulate()` method.

Let me know if you have any trouble getting the same result for different runs.

**Comment by jasmainak** [comment]
>  it's only because we get different values of yr, and yt when we call the simulate() method.

should we also make this accept a `random_state`? I don't see any reason to have it as random. It will make debugging harder

**Comment by pavanramkumar** [comment]
@jasmainak It already does. `simulate()` draws poisson random variables using `numpy.random.poisson` to simulate spike counts. It sets the seed using `self.random_state`. If you change `random_state` you will get different `yr`, `yt`.

**Comment by jasmainak** [comment]
> As a results, the default learning_rate has increased. The API examples might not have been updated since then, but try using the default learning rates. 

@pavanramkumar should we update these examples then? It might be confusing for a beginner that the website examples don't work as expected.

**Comment by yves-weissenberger** [comment]
Hi,

thanks for the quick responses, problem is fixed :).

![image](https://cloud.githubusercontent.com/assets/9501913/16710416/2b0dafc4-4625-11e6-8ba1-f622ef41c38c.png)

As an end user, I think I would recommend changing the examples. I came across this package after having fit my GLMs with scipy, statsmodels and lucas theis cmt module. Even then I was confused, assumed it was some dependency problem. 

As a side note, I'm happy to share performance benchmarks on my data, of different fitting methods, once its ready (soon...); if thats helpful?

**Comment by pavanramkumar** [comment]
@yves-weissenberger glad it's sorted out for now! and cool to hear about your journey to `pyglmnet` :)

yes, i'm working on updating the examples right away.

we're working on organizing the examples gallery into three categories: 
- API demos 
- benchmarks (i.e. time complexity and accuracy comparisons against `R` and `scikit-learn`)
- applications (demos of the package being applied to real data)

working on all of those at present, but we absolutely welcome your suggestions and PRs.

**Comment by pavanramkumar** [comment]
ok, i've updated the example api pages to reflect the default learning rates. i will close this issue. thanks for reporting, it leads to epsilon improvements.
